components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/ApplicationConfig.java

```diff
@@ -274,4 +308,30 @@ public Object clone() throws CloneNotSupportedException {
         applicationConfig.setPermissions(this.permissions.clone());
         return applicationConfig;
     }
+
+    /**
+     * Set all the claim mappings of the configured SP claim dialects.
+     *
+     * @param claimConfig Application claim configuration
+     * return Application claim mappings
+     */
+    private void setSpDialectClaims(ClaimConfig claimConfig, List<ClaimMapping> spClaimMappings) {
+        String[] spClaimDialects = claimConfig.getSpClaimDialects();
+        if(!ArrayUtils.isEmpty(spClaimDialects)) {
+            ClaimMetadataManagementService claimMetadataMgtService = new ClaimMetadataManagementServiceImpl();
```

Comment:
This directly binds to the available service implementation.
Either the framework should listen to the OSGi service and get hold of it or I see some such functionalities are provided from the 'ClaimMetadataHandler' of claim.metadata component wrapping the service. So either the API addition should be included there and it should be invoked from this point.
Do the appropriate.